Add missing attributes to `Repository`

RobPasMue opened the pull request:
According to https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#update-a-repository

New parameters are accepted and others have been removed.

Parameters included in this PR:
``visibility``
``is_template``
``use_squash_pr_title_as_default``
``squash_merge_commit_title``
``squash_merge_commit_message``
``merge_commit_title``
``merge_commit_message``
``web_commit_signoff_required``

Paramters removed in this PR:
``has_downloads``

Parameters have also been ordered according to the REST API order in the documentation

RobPasMue commented:
Hi @trim21 ! Let me know if I need to do anything else to get this feature merged :) I hope I did things properly

EnricoMi commented on github/Repository.py:
```diff
@@ -1772,6 +1774,7 @@ def edit(
         assert description is github.GithubObject.NotSet or isinstance(description, str), description
         assert homepage is github.GithubObject.NotSet or isinstance(homepage, str), homepage
         assert private is github.GithubObject.NotSet or isinstance(private, bool), private
+        assert visibility is github.GithubObject.NotSet or isinstance(visibility, str), visibility
```
Can we assert here `visibility` has only the allowed values?
```suggestion
        assert visibility is github.GithubObject.NotSet or visibility in [
            "public", "private"
        ], visibility
```

EnricoMi reviewed (changes requested):
While you are adding a missing attribute, can you add those that are also missing, e.g.
- is_template
- use_squash_pr_title_as_default
- squash_merge_commit_title
- squash_merge_commit_message
- merge_commit_title
- merge_commit_message
- web_commit_signoff_required

Please check if I missed some.

RobPasMue commented:
Yep, for sure @EnricoMi - let me do that! =)

EnricoMi commented on tests/Repository.py:
```diff
@@ -134,7 +134,6 @@ def testEditWithAllArguments(self):
             has_issues=True,
```
Can we test will all new arguments? Recording test data should be easy to do.

EnricoMi reviewed (changes requested):
Can we get add all those new attributes to `_initAttributes` and `_useAttributes` and create getters as well? Please record test data for `testAttributes` and add those attributes there as well.

RobPasMue commented:
Hi @EnricoMi - issues addressed. I added the properties, initialized the attributes and modified the useAttributes function. I also updated the test data. I hope everything is fine now. Let me know if you need anything else

MaxJPRey commented:
Nice work @RobPasMue . Thanks for updating the properties according to the API.

EnricoMi reviewed (approved):
Awesome, good work!
